retry.go

```diff
@@ -94,13 +88,11 @@ func XWithContext(ctx context.Context, x int, maxBackoff time.Duration, f func(c
 				return nil
 			}
 		}
+
+		timer.Reset(backoff(i+1, maxBackoff))
 	}
 	// ran out of retries
-	if latestErr != nil {
-		return fmt.Errorf("%w", latestErr)
```

Comment:
Could you retain this wrapped error? It encourages people to use `errors.Is` and will allow us to return more context about a failure in the future.

Reply:
Oh, you want to use `%w` to prevent folks from doing `if err == someOtherError`? I'm fine with that.